src/utility/Gui/Gui.cpp

```diff
@@ -195,12 +198,17 @@ void Gui::keyboard(const char* title, char* text, uint8_t length) {
 				cursorX = 3;
 			} else if (cursorY == 4 && cursorX < 10) {
 				cursorX = 10;
+			} else if (cursorX == 12 || (cursorX >=10 && (cursorY == 0 || cursorY == 4)) {
+				cursorX = 0;  // Wrap around
 			} else {
 				cursorX++;
 			}
 		}
-		cursorX = min(12, max(0, cursorX));
-		cursorY = min(4, max(0, cursorY - gb.buttons.repeat(BUTTON_UP, 4) + gb.buttons.repeat(BUTTON_DOWN, 4)));
+		// cursorY movement
+		cursorY += gb.buttons.repeat(BUTTON_DOWN, 4) - gb.buttons.repeat(BUTTON_UP, 4);
+		if (cursorY == 5) cursorY = 0;
```

Comment:
for wrapping stuff it is always good practice to use `>=` or `>` or `<` or `<=` rather than `==`, because, if you have the greater / less than comparison, your program won't break if, due to another bug, your cursor suddenly jumps by two

Reply:
(same of course for the if-condition right after)